code/modules/shuttle/supply.dm

```diff
@@ -108,131 +115,543 @@
-		credits_to_deposit += credits_from_intel
-
-	if(crate_count > 0)
-		msg += "<span class='good'>+[total_crate_value]</span>: Received [crate_count] crate(s).<br>"
-		credits_to_deposit += total_crate_value
+	var/list/credit_changes = list()
+	for(var/datum/economy/line_item/item in manifest.line_items)
+		if(!credit_changes[item.account])
+			credit_changes[item.account] = 0
+		credit_changes[item.account] += item.credits
+
+		if (item.credits > 0)
+			msg += "<span class='good'>[item.account.account_name] +[item.credits]</span>: [item.reason]<br>"
+		else
+			msg += "<span class='bad'>[item.account.account_name] [item.credits]</span>: [item.reason]<br>"
+
+	for(var/account in credit_changes)
+		if(credit_changes[account] > 0)
+			GLOB.station_money_database.credit_account(account, credit_changes[account], "Supply Shuttle Exports Payment", "Central Command Supply Master", supress_log = FALSE)
+		else
+			GLOB.station_money_database.charge_account(account, -credit_changes[account], "Supply Shuttle Fine", "Central Command Supply Master", allow_overdraft = TRUE, supress_log = FALSE)
 
 	SSeconomy.centcom_message += "[msg]<hr>"
-	if(credits_to_deposit > 0)
-		GLOB.station_money_database.credit_account(SSeconomy.cargo_account, credits_to_deposit, "Supply Shuttle Exports Payment", "Central Command Supply Master", supress_log = FALSE)
-		if(research_credits)
-			GLOB.station_money_database.credit_account(GLOB.station_money_database.get_account_by_department(DEPARTMENT_SCIENCE), research_credits, "Supply Shuttle Exports Payment", "Central Command Supply Master", supress_log = FALSE)
-		if(service_credits)
-			GLOB.station_money_database.credit_account(GLOB.station_money_database.get_account_by_department(DEPARTMENT_SERVICE), service_credits, "Supply Shuttle Exports Payment", "Central Command Supply Master", supress_log = FALSE)
+	manifest = new
+
+
+// Convenience object that registers itself with the supply shuttle and provides
+// methods for you to override.
+/datum/economy/simple_seller
+
+/datum/economy/simple_seller/proc/register(obj/docking_port/mobile/supply/S)
+	RegisterSignal(S, COMSIG_CARGO_BEGIN_SCAN,			PROC_REF(begin_scan))
+	RegisterSignal(S, COMSIG_CARGO_CHECK_SELL,			PROC_REF(check_sell))
+	RegisterSignal(S, COMSIG_CARGO_BEGIN_SELL,			PROC_REF(begin_sell))
+	RegisterSignal(S, COMSIG_CARGO_DO_PRIORITY_SELL,	PROC_REF(sell_priority))
+	RegisterSignal(S, COMSIG_CARGO_DO_SELL,				PROC_REF(sell_normal))
+	RegisterSignal(S, COMSIG_CARGO_SEND_ERROR,			PROC_REF(sell_wrong))
+	RegisterSignal(S, COMSIG_CARGO_END_SELL,			PROC_REF(end_sell))
+
+/datum/economy/simple_seller/proc/begin_scan(obj/docking_port/mobile/supply/S)
+	SIGNAL_HANDLER
+	return
+
+/datum/economy/simple_seller/proc/check_sell(obj/docking_port/mobile/supply/S, atom/movable/AM)
+	SIGNAL_HANDLER
+	return NONE
+
+/datum/economy/simple_seller/proc/begin_sell(obj/docking_port/mobile/supply/S)
+	SIGNAL_HANDLER
```

Comment:
Could you comment the signals that these are responding to next to the SIGNAL_HANDLER usages? It's not mandatory but especially for something like this I think it'd help make clear what signal it corresponds to